lib/puppet.rb

```diff
@@ -138,14 +138,14 @@ def self.run_mode
   # @api public
   # @param args [Array<String>] the command line arguments to use for initialization
   # @return [void]
-  def self.initialize_settings(args = [])
-    do_initialize_settings_for_run_mode(:user, args)
+  def self.initialize_settings(args = [], require_config = true)
```

Comment:
At one point when we were talking about this, there was talk of having `require_config` be a named argument in the places where it wasn't the only method argument. Decided against it in the end?

Reply:
I think it wasn't working at first, so I just did this. I'm happy to change it to a named argument if that's cleaner/preferred

Reply:
I generally prefer it for anything with optional arguments, but I'm not sure that it matters that much in this particular case? It's a bit awkward here, since it would be ideal to have both `args`, and `require_config` be named args, as they're both optional, but that'd be an API. change that I don't think we could actually do. Since we can't make them both named, I'm not sure it is sufficiently worth it?

Reply:
turn out, we still need to support ruby 1.9.3 in the 5.5.x branch, so named args aren't an option for this PR

Reply:
👍 but also 😔